worker/uniter/uniter.go

```diff
@@ -182,15 +182,18 @@ func newUniter(uniterParams *UniterParams) func() (worker.Worker, error) {
 		applicationChannel:   uniterParams.ApplicationChannel,
 	}
 	startFunc := func() (worker.Worker, error) {
-		if err := catacomb.Invoke(catacomb.Plan{
+		plan := catacomb.Plan{
 			Site: &u.catacomb,
 			Work: func() error {
 				return u.loop(uniterParams.UnitTag)
 			},
-			Init: []worker.Worker{
+		}
+		if u.modelType == model.CAAS {
+			plan.Init = []worker.Worker{
```

Comment:
what about

// For CAAS models, make sure the leadership tracker is killed when the Uniter
// dies
plan.Init = append(plan.Init, u.leadershipTracker)

Reply:
yep, thx